### PR TITLE
Fixes 3070 - Autosuggest starts at 3 characters

### LINKED ISSUE #3070: Start autocomplete search after n chars entered
The autocomplete should only start sending requests once there's more than `n` characters because the results for a single char aren't that useful.

In the past typical values for `n` would be 2-3 chars - it'll make sense to make this configurable so it's easily tweaked. Let's start with 2 chars and go from there.

### PR DESCRIPTION
**Pull request by hellokarthic**
Fixes #3070 

**Before**:

image

**After**:

image

### CODE REVIEW

File: src/amo/components/SearchForm/index.js [hunk 1]
```diff
@@ -189,6 +189,10 @@ export class SearchFormBase extends React.Component {
     this.props.router.push(suggestion.url);
   }
 
+  shouldRenderSuggestions= (value) => {
```
Comment (tofumatt) [inline src/amo/components/SearchForm/index.js]: Nitpick but there should be a space before the `=` here, after `shouldRenderSuggestions`.

File: tests/unit/amo/components/TestSearchForm.js [hunk 2]
```diff
@@ -319,8 +319,8 @@ describe(__filename, () => {
 
       sinon.assert.notCalled(dispatch);
 
-      // User types 'a' in the search input.
-      wrapper.find('input').simulate('change', createFakeChangeEvent('a'));
+      // User types 'abc' in the search input.
```
Comment (tofumatt) [inline tests/unit/amo/components/TestSearchForm.js]: Hah, thanks for updating the comment 😄

**Comment by hellokarthic** [comment]
@tofumatt Hi, Can you help me write test cases for this. I've no previous experience in this. But I'd love to learn :) 
Although I tried to figure it out on my own. Check the below test case. But this didn't help!

> it('does not show auto suggestions for less than min search parameters', () => {
>       const fakeDispatch = sinon.stub();
>       const wrapper = mountBaseComponent({
>         dispatch: fakeDispatch,
>         query: 'a'.repeat(2),
>       });
>       expect(wrapper.state('autocompleteIsOpen')).toEqual(false);
>     });

**Comment by tofumatt** [comment]
Hi there,

I will be able to help but I’m attending a conference today and wrapping up a few issues of my own before I take a week off, so I won’t be able to help anytime soon :-(

Sorry about that.

If anyone else sees this and can help feel free; otherwise I’ll take a look when I’m back.

**Comment by tofumatt** [comment]
Hi there!

Looks like there are some conflicts now that need to be resolved.

Sorry about not getting to this but we've been pretty busy with React upgrades and some personal goals for the quarter. I don't think I'll be able to give any advice/help on writing tests beyond what's in our code. Our tests are using jest/enzyme so if you searched for how to write tests or looked through our code that would be the best place to start I think.

**Comment by muffinresearch** [comment]
@hellokarthic since it's been a while since there's been an update to this I'm going to close it. But please do re-open when it's ready for a re-review.

**Pull request closed without merging**